windows-agent/cmd/ubuntu-pro-agent/agent/agent.go

```diff
@@ -221,7 +228,7 @@ func (a *App) publicDir(opts options) (string, error) {
 		opts.publicDir = filepath.Join(homeDir, common.UserProfileDir)
 	}
 
-	if err := os.MkdirAll(opts.publicDir, 0600); err != nil {
+	if err := os.MkdirAll(opts.publicDir, 0700); err != nil {
```

Comment:
Yeah, it didn't matter until now because it wasn't properly tested. On windows this argument is ignored, this is the function that actually creates the directory in the syscall package:

```
func Mkdir(path string, mode uint32) (err error) {
	pathp, err := UTF16PtrFromString(path)
	if err != nil {
		return err
	}
	return CreateDirectory(pathp, nil)
}
```

Now that this is tested, the Linux tests were failing with `permission denied` specifically because the log file could not be created inside the public dir.